Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -425,6 +441,21 @@ public object Deserialize(IIonReader reader, Type type, IonType ionType)
 
             if (ionType == IonType.Struct) 
             {
+                if (type.IsGenericType && type.GetGenericTypeDefinition().IsAssignableFrom(typeof(Dictionary<,>)))
+                {
+                    var genericArguments = type.GetGenericArguments();
+                    if (genericArguments[0].IsAssignableTo(typeof(string)))

```

Comment:
Minor nit: can the valid/invalid if/else ordering be kept consistent? In the Serialize it's (if invalid else valid), while here it's (if valid else invalid).

Reply:
Will fix!